**  Fichiers modifiées 6.5.0 **

/adherent/consulter_cotisations_adht.php
/adherent/export_tadhts.php
/adherent/export_tcotis.php
/adherent/remplir_cotisations_adht.php
/admin/upgrade640-650.php
/config/lang.php
/config/version.php  
/doc/Changelog.txt
/doc/lisez_moi.txt 
/doc/tables_desciption.rtf
/install/schema.php
/templates/adherent/consulter_cotisations_adht.tpl
/templates/adherent/remplir_cotisations_adht.tpl
/-------LIB-------------
/lib/version_lib_smarty.txt 
/lib/smarty/version.txt
/lib/smarty/Smarty.class.php
/lib/smarty/change_log.txt
/lib/smarty/sysplugins/smarty_security.php
/lib/smarty/sysplugins/smarty_resource.php
/lib/smarty/sysplugins/smarty_internal_utility.php
/lib/smarty/sysplugins/smarty_internal_templatecompilerbase.php
/lib/smarty/sysplugins/smarty_cacheresource.php
/lib/smarty/sysplugins/smarty_internal_cacheresource_file.php
/lib/smarty/sysplugins/smarty_internal_compile_block.php
/libsmarty/sysplugins/smarty_internal_template.php

// - 6.4.0 05/04/2013 Finale




30/06/2013
---------
523a6169987fce687647423419300cd3fbb88ff0
 .../adherent/consulter_cotisations_adht.php        |   6 +-
 gestion/gestassophp_sa/adherent/export_tadhts.php  |   2 +-
 gestion/gestassophp_sa/adherent/export_tcotis.php  |  12 +-
 .../adherent/remplir_cotisations_adht.php          |  59 +--
 gestion/gestassophp_sa/admin/upgrade640-650.php    |  73 ++--
 gestion/gestassophp_sa/config/lang.php             |   6 +-
 gestion/gestassophp_sa/doc/Changelog.txt           |  14 +-
 gestion/gestassophp_sa/doc/tables_desciption.rtf   | 210 ++++-----
 gestion/gestassophp_sa/install/schema.php          |   1 +
 .../adherent/consulter_cotisations_adht.tpl        |  11 +-
 .../adherent/remplir_cotisations_adht.tpl          |  11 +
 gestion/lib/smarty/Smarty.class.php                |  17 +-
 gestion/lib/smarty/change_log.txt                  |  38 +-
 .../lib/smarty/sysplugins/smarty_cacheresource.php |   2 +-
 .../smarty_internal_cacheresource_file.php         | 469 +++++++++++----------
 .../sysplugins/smarty_internal_compile_block.php   |   5 +-
 .../smarty/sysplugins/smarty_internal_template.php |   9 +-
 .../smarty_internal_templatecompilerbase.php       |  20 +-
 .../smarty/sysplugins/smarty_internal_utility.php  |  14 +-
 gestion/lib/smarty/sysplugins/smarty_resource.php  |   4 +-
 gestion/lib/smarty/sysplugins/smarty_security.php  |   2 +-
 gestion/lib/smarty/version.txt                     |   1 +
 gestion/lib/version_lib_smarty.txt                 |   3 +-
 23 files changed, 523 insertions(+), 466 deletions(-)

diff --git a/gestion/gestassophp_sa/adherent/consulter_cotisations_adht.php b/gestion/gestassophp_sa/adherent/consulter_cotisations_adht.php
index 9fb3242..e265fe9 100644
--- a/gestion/gestassophp_sa/adherent/consulter_cotisations_adht.php
+++ b/gestion/gestassophp_sa/adherent/consulter_cotisations_adht.php
@@ -51,9 +51,10 @@ $pas = $logpass[1];
 	/***** Si ADMINISTRATEUR donc $priorite_adht >4  DROIT DE CONSUTER ET MODIFIER     (4 n'a PAS le droit)	*/
 	if ($priorite_adht > 5 ) { // AUTORISATION
 		$id_cotis_adht = get_post_variable_numeric('id_cotis', '');// l'id de la cotisation existante		
-		$affiche_message =' N° '.$id_cotis_adht;	
+		$affiche_message =' N&deg; '.$id_cotis_adht;	
 		$req_lire_info_cotis = "SELECT id_cotis,qui_cotis,id_adhtasso,"
 		."id_type_cotis,montant_cotis,info_cotis,date_enregist_cotis,"
+		."paiement_cotis,"  //+ Ajout Zone PAIEMENT
 		."date_debut_cotis,date_fin_cotis,info_archiv_cotis,datemodiffiche_cotis,"
 		." id_type_cotisation,nom_type_cotisation" // TABLE_TYPE_COTISATIONS
 		." FROM ".TABLE_COTISATIONS.", ".TABLE_TYPE_COTISATIONS 
@@ -65,7 +66,8 @@ $pas = $logpass[1];
 		// le nom de la  cotisation  = $cotis_adh[nom_type_cotisation]			
 		$cotis_adh['date_enregist_cotis'] = switch_sqlFr_date($cotis_adh['date_enregist_cotis']);
 		$cotis_adh['date_debut_cotis'] = switch_sqlFr_date($cotis_adh['date_debut_cotis']);
-		$cotis_adh['date_fin_cotis'] = switch_sqlFr_date($cotis_adh['date_fin_cotis']);			
+		$cotis_adh['date_fin_cotis'] = switch_sqlFr_date($cotis_adh['date_fin_cotis']);	
+		$cotis_adh['paiement_cotis'] = $T_PAIEMENT_COTIS[$cotis_adh['paiement_cotis']];	//+ Ajout Zone PAIEMENT	
 			
 		// les données perso
 		$req_lire_perso_adht = "SELECT civilite_adht, nom_adht, prenom_adht, adresse_adht,"
diff --git a/gestion/gestassophp_sa/adherent/export_tadhts.php b/gestion/gestassophp_sa/adherent/export_tadhts.php
index bcdbce6..930feea 100644
--- a/gestion/gestassophp_sa/adherent/export_tadhts.php
+++ b/gestion/gestassophp_sa/adherent/export_tadhts.php
@@ -58,7 +58,7 @@ if (($sessionadherent) && $log == ($_SESSION['ses_login_adht']) && $pas == ($_SE
 	." FROM  ".TABLE_ADHERENTS.", ".TABLE_ANTENNE." WHERE antenne_adht=id_type_antenne ";
 	
 // 1ere ligne
-print("Num\t Societaire\t "._LANG_FICHE_ADHT_CIVIL."\t "._LANG_FICHE_ADHT_PRENOM."\t "._LANG_TPL_COL_ADHT_NOM."\t "._LANG_FICHE_ADHT_ADRESS."\t "._LANG_FICHE_ADHT_CP."\t "._LANG_TPL_COL_ADHT_VILLE."\t "._LANG_TPL_COL_ADHT_TELEPH."\t "._LANG_TPL_COL_ADHT_PORTABLE."\t "._LANG_FICHE_ADHT_FAX."\t "._LANG_FICHE_ADHT_MAIL."\t DateCreationFiche\t "._LANG_FICHE_ADHT_ANT."\t "._LANG_TPL_ADHT_DATENAIS."\t visible\t "._LANG_VISU_FICHE_ADHT_LASTMOD."\t  "._LANG_FICHE_ADHT_WEB."\t "._LANG_ADMIN_PRIORITE_COL_PRIORITE."\t "._LANG_FICHE_COTIS_ADHT_DATE_FIN."\t Date_Sortie\t "._LANG_FICHE_ADHT_TAGE."\t "._LANG_FICHE_ADHT_PROMOTION."\t "._LANG_FICHE_ADHT_FICHE_ENR."\t "._LANG_FICHE_ADHT_COMPL."\n");
+print("Num\t Societaire\t Civilite\t Prenom\t "._LANG_TPL_COL_ADHT_NOM."\t "._LANG_FICHE_ADHT_ADRESS."\t "._LANG_FICHE_ADHT_CP."\t "._LANG_TPL_COL_ADHT_VILLE."\t Telephone\t "._LANG_TPL_COL_ADHT_PORTABLE."\t "._LANG_FICHE_ADHT_FAX."\t "._LANG_FICHE_ADHT_MAIL."\t DateCreationFiche\t "._LANG_FICHE_ADHT_ANT."\t "._LANG_TPL_ADHT_DATENAIS."\t visible\t Derniere modif\t  "._LANG_FICHE_ADHT_WEB."\t Priorite\t "._LANG_FICHE_COTIS_ADHT_DATE_FIN."\t Date_Sortie\t Tranche age\t "._LANG_FICHE_ADHT_PROMOTION."\t Qui enregistre Fiche\t "._LANG_FICHE_ADHT_COMPL."\n");
 
 // Boucle préparation écriture de la ligne	
 	$dbresult = $db->Execute($req_lire_table_adht);
diff --git a/gestion/gestassophp_sa/adherent/export_tcotis.php b/gestion/gestassophp_sa/adherent/export_tcotis.php
index 4a3b60a..73d7b2f 100644
--- a/gestion/gestassophp_sa/adherent/export_tcotis.php
+++ b/gestion/gestassophp_sa/adherent/export_tcotis.php
@@ -51,10 +51,11 @@ if (($sessionadherent) && $log == ($_SESSION['ses_login_adht']) && $pas == ($_SE
      
     echo("$title\n"); // affiche la ligne de titre
 
-	//id_cotis	qui_cotis	id_adhtasso	id_type_cotis	montant_cotis	info_cotis	date_enregist_cotis	date_debut_cotis	date_fin_cotis	cotis	info_archiv_cotis	datemodiffiche_cotis
+	//id_cotis	qui_cotis	id_adhtasso	id_type_cotis	montant_cotis	info_cotis	date_enregist_cotis	date_debut_cotis	date_fin_cotis	cotis	+ paiement_cotis info_archiv_cotis	datemodiffiche_cotis export_tcotis.php
 	
 	$req_lire_table_cotis = "SELECT id_cotis,id_adhtasso,id_type_cotis,"
 	." montant_cotis,info_cotis,date_enregist_cotis,date_debut_cotis,date_fin_cotis,cotis,"
+	."paiement_cotis,"  //+ Ajout Zone PAIEMENT
 	." info_archiv_cotis,datemodiffiche_cotis,"
 	." prenom_adht,nom_adht," // TABLE_ADHERENTS
 	." nom_type_cotisation" // TABLE_TYPE_COTISATIONS
@@ -63,7 +64,7 @@ if (($sessionadherent) && $log == ($_SESSION['ses_login_adht']) && $pas == ($_SE
 	." AND ".TABLE_TYPE_COTISATIONS.".id_type_cotisation=".TABLE_COTISATIONS."
 	.id_type_cotis ORDER BY id_cotis";	
 		
-    print("Num\t Nom Prenom\t type\t montant\t information\t date_enregist\t date_debut\t date_fin\t archive\t info_archiv\t date_modif_fiche\n");
+    print("Num\t Nom Prenom\t type\t montant\t information\t date_enregist\t date_debut\t date_fin\t archive\t paiement\t info_archiv\t date_modif_fiche\n");
 	
 	$dbresult = $db->Execute($req_lire_table_cotis);	
 	
@@ -81,15 +82,16 @@ if (($sessionadherent) && $log == ($_SESSION['ses_login_adht']) && $pas == ($_SE
 		if ($champ9 =='999') {
 			$champ9 = 'Archive';// Affichage du statut Archivée
 		}
-		$champ10 = $row['info_archiv_cotis']; // 	
-		$champ11 = switch_sqlFr_date($row['datemodiffiche_cotis']);
+		$champ10 = $row['paiement_cotis']; //+ Ajout Zone PAIEMENT 	 
+		$champ11 = $row['info_archiv_cotis']; 	
+		$champ12= switch_sqlFr_date($row['datemodiffiche_cotis']);
 		if ($champ11 == '00/00/0000'){
 			$champ11='';
 		}					
 
 		//ecriture de la ligne
 		print ("$champ0\t $champ2\t $champ3\t $champ4\t $champ5\t $champ6\t $champ7\t"
-		." $champ8\t $champ9\t $champ10\t $champ11 \n");
+		." $champ8\t $champ9\t $champ10\t $champ11\t $champ12 \n");
 		
 	}
 
diff --git a/gestion/gestassophp_sa/adherent/remplir_cotisations_adht.php b/gestion/gestassophp_sa/adherent/remplir_cotisations_adht.php
index e722bd4..ff95aed 100644
--- a/gestion/gestassophp_sa/adherent/remplir_cotisations_adht.php
+++ b/gestion/gestassophp_sa/adherent/remplir_cotisations_adht.php
@@ -284,7 +284,7 @@ $pas = $logpass[1];
 		if ($cotis_adh['id_type_cotisation'] =='') {
 			$erreur_saisie ['type_cotisation'] = _LANG_MESSAGE_COTIS_ADHT_ALERT_TYPE;
 		}	
-//++
+
 		// si création le montant cotisation est affecté automatiquement 
 		if  (isset( $required ['creation_cotisation']) and $required ['creation_cotisation'] == 1) {
 		//if  ($required ['creation_cotisation'] == 1)  {
@@ -302,7 +302,7 @@ $pas = $logpass[1];
 			if ($id_listemontant_cotisation >=1) {
 			$cotis_adh['montant_cotis'] = $tab_montant_cotisation[$id_listemontant_cotisation];
 			}
-//++	
+
 			} else { // Montant cotisation
 			$cotis_adh['montant_cotis']=(get_post_variable('montant_cotis',''));
 			if ( ($cotis_adh['montant_cotis'] =='') || (!is_numeric($cotis_adh['montant_cotis'])) ) {
@@ -334,7 +334,9 @@ $pas = $logpass[1];
 		} else { // si date vide
 			$erreur_saisie ['d_fin_cotis'] = _LANG_MESSAGE_LISTE_COTIS_ADHT_DATE_FIN;
 		}
-		
+		// Ajout Zone PAIEMENT
+		$cotis_adh['paiement_cotis']=((get_post_variablehtml('paiement_cotis','')));
+
 		// Commentaire de cotisation
 		$cotis_adh['info_cotis']= stripslashes((get_post_variablehtml('info_cotis',''))); // elnlève \  si on a fait une erreur
 	
@@ -359,45 +361,16 @@ $pas = $logpass[1];
 				$erreur_saisie ['id_adhtasso'] = _LANG_MESSAGE_COTIS_ADHT_ALERT_NOM.' '.ADHERENT_BENE;
 			}
 			
-					
-			/***** Verifier S'il n' y pas UNE AUTRE Cotis en cours dans la BD  pour une Id_AdhtAsso  */
-/*			
-			// counter le Nb de id_cotis avec  qui_cotis=adh	id_adhtasso=  Id du Cotisant
-			$requete[1] = "SELECT id_cotis FROM "
-			.TABLE_COTISATIONS." WHERE qui_cotis ='adh' "	
-			." AND cotis='' AND	id_adhtasso='$cotis_adh[id_adhtasso]' ";
-			
-			// comptage des fiches		
-			$dbresult = $db->Execute($requete[1]); //Pour compter le NB d'enregistrements
-//test si aucune cotisation de saisie b2+
-			if ($dbresult) {
-				$nb_cotis = $dbresult->RecordCount() ; //Pour compter le NB d'enregistrements	
-			}else {
-				$nb_cotis = 0 ;
-			}			
-			if ($nb_cotis >= 1) {
-				$alert_saisie['id_adhtasso'] = $nb_cotis ;
-				// voir  les N°
-					while ($dbresult && $row = $dbresult->FetchRow()) {						
-					//affiche les variables de la ligne 
-					$num_id_cotis[$indice]['id_cotis'] = $row['id_cotis']; //Nom fichier					
-					$indice++;
-				}
-
-				$tpl->assign('num_id_cotis',$num_id_cotis); // tableau 			
-			
-	
-			}
-	*/		
-
 			// Si Aucune erreur de saisie ON Valide --> Requette enregistrement nouvelle cotisation	
 			if (count($erreur_saisie) == 0) {
 				$cotis_adh['info_cotis'] = addslashes($cotis_adh['info_cotis']); // ajoute \ si on a fait une erreur
 				$req_ecrit_nouvelle_cotis = "INSERT INTO ".TABLE_COTISATIONS
-				." (id_adhtasso, qui_cotis, id_type_cotis, montant_cotis, info_cotis," 
+				." (id_adhtasso, qui_cotis, id_type_cotis, montant_cotis, info_cotis,"  
+				." paiement_cotis," //+ Ajout Zone PAIEMENT
 				." date_enregist_cotis, date_debut_cotis, date_fin_cotis)" 					
 				." VALUES('$cotis_adh[id_adhtasso]','adh','$cotis_adh[id_type_cotisation]',"
-				."'$cotis_adh[montant_cotis]','$cotis_adh[info_cotis]',"
+				."'$cotis_adh[montant_cotis]','$cotis_adh[info_cotis]',"  
+				."'$cotis_adh[paiement_cotis]',"  //+ Ajout Zone PAIEMENT
 				." '$cotis_adh[date_enregist_cotis_sql]','$cotis_adh[date_debut_cotis_sql]',"
 				."'$cotis_adh[date_fin_cotis_sql]')";
 				$dbresult = $db->Execute($req_ecrit_nouvelle_cotis);				
@@ -430,17 +403,19 @@ $pas = $logpass[1];
 			
 			// Si Aucune erreur de saisie Udpate --> Requette enregistrement update cotisation				
 			if (count($erreur_saisie)==0) {	  
-				$cotis_adh[info_cotis] = addslashes($cotis_adh[info_cotis]); // ajoute \ si on a fait une erreur
+				$cotis_adh['info_cotis'] = addslashes($cotis_adh['info_cotis']); // ajoute \ si on a fait une erreur
 				$req_ecrit_modif_cotis =("UPDATE ".TABLE_COTISATIONS
 				." SET id_type_cotis='$cotis_adh[id_type_cotisation]',"
 				." montant_cotis= '$cotis_adh[montant_cotis]', "
-				." info_cotis='$cotis_adh[info_cotis]',"
+				." info_cotis='$cotis_adh[info_cotis]',"  
+				." paiement_cotis='$cotis_adh[paiement_cotis]'," // + Ajout Zone PAIEMENT
 				." date_debut_cotis='$cotis_adh[date_debut_cotis_sql]',"	
 				." date_fin_cotis='$cotis_adh[date_fin_cotis_sql]',"
 				." datemodiffiche_cotis='$date_du_jour'"
 				." WHERE id_cotis='$id_cotis_adht'"); 
 				$dbresult = $db->Execute($req_ecrit_modif_cotis);	
-				
+
+			
 				/***** mise A JOUR DE la table adherent  Date_Echeance_Cotis  */
 				$req_ecrit_nouvelle_cotis_adht =("UPDATE ".TABLE_ADHERENTS
 				." SET date_echeance_cotis='$cotis_adh[date_fin_cotis_sql]'"
@@ -471,7 +446,8 @@ $pas = $logpass[1];
 	// préparation des données pour affichage si une cotisation existe déja
 		if ($required ['modification_cotisation'] == 1) {
 			$req_lire_info_cotis = "SELECT id_cotis,qui_cotis,id_adhtasso,"
-			."id_type_cotis,montant_cotis,info_cotis,date_enregist_cotis,"
+			."id_type_cotis,montant_cotis,info_cotis,date_enregist_cotis,"  
+			."paiement_cotis,"  //+ Ajout Zone PAIEMENT
 			."date_debut_cotis,date_fin_cotis,info_archiv_cotis,datemodiffiche_cotis,"
 			." id_type_cotisation,nom_type_cotisation" // TABLE_TYPE_COTISATIONS
 			." FROM ".TABLE_COTISATIONS.", ".TABLE_TYPE_COTISATIONS 
@@ -487,7 +463,7 @@ $pas = $logpass[1];
 				$cotis_adh['date_fin_cotis'] = switch_sqlFr_date($cotis_adh['date_fin_cotis']);			
 				// Préparation pour Affichage partie variable en fonction des données
 				$tpl->assign('data_cotis_adh',$cotis_adh);
-				$tpl->assign('modif_fiche',$required ['modification_cotisation']);			
+				$tpl->assign('modif_fiche',$required ['modification_cotisation']);	
 			}
 		} 
 			
@@ -535,6 +511,7 @@ $pas = $logpass[1];
 	$tpl->assign('required',$required); // Variables Obligatoires
 	$tpl->assign('erreur_saisie',$erreur_saisie); // Erreur de saisie sur champs Obligatoires
 	$tpl->assign('alert_saisie',$alert_saisie); // Message alerte	
+	$tpl->assign('list_paiement_cotis',$T_PAIEMENT_COTIS);//+  Ajout Zone PAIEMENT Gestion Cotisations	
 	
 	$tpl->assign('disabled',$disabled); // pour afficher "disabled" les zones non modifiables du formulaire
 	$tpl->assign('affiche_message',$affiche_message); // pour afficher
diff --git a/gestion/gestassophp_sa/admin/upgrade640-650.php b/gestion/gestassophp_sa/admin/upgrade640-650.php
index d05d744..cacedd7 100644
--- a/gestion/gestassophp_sa/admin/upgrade640-650.php
+++ b/gestion/gestassophp_sa/admin/upgrade640-650.php
@@ -17,6 +17,7 @@
  *  Directory :  /ROOT_DIR_GESTASSO/admin/
  *   Fichier : upgrade5-sa.php
  *   MISE A JOUR ENTRE V 5.x et GestAssoPhp+Pg (gestassophp_sa)
+ *   upgrade depuis 6.4.0  vers 6.5.0
 */
 
 error_reporting(0);
@@ -35,64 +36,54 @@ echo '
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
 <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="fr">
 <head>
-    <title>GestAssoPhp+Pg mise à jour de V 6.4.0 vers 6.5.0(gestassophp_sa)</title>
+    <title>GestAssoPhp+Pg mise &agrave; jour de V 6.4.0 vers 6.5.0(gestassophp_sa)</title>
     <meta http-equiv="Content-Type" content="text/HTML; charset={language name=charset}" />		
 	<link rel="stylesheet" type="text/css" media="screen"  href="../js/style_screen.css"/>
 </head>
 <body> 
 ';
-
+	//http://phplens.com/lens/adodb/docs-datadict.htm
+	$dbdict = NewDataDictionary($db);
+		
     echo 'GestAssoPhp+Pg mise &agrave; jour de V 6.4.0 GestAssoPhp vers '.$VERSION.' (gestassophp_sa) le : '.$date_du_jour.'  <br /><br />';
 	
 	echo '<big>MODIFICATION BASE DE DONNEES </big><br />';
 	echo '------------------------------------------- <br />';
-	echo '<b>Modifications Champs cp_adht dans table '.TABLE_ADHERENTS.'</b><br />pour afficher le code postal commen&ccedil;ant par 0 <br />';	
+	
+	echo '<b>1- Modifications Champs cp_adht dans table '.TABLE_ADHERENTS.'</b><br />pour afficher le code postal commen&ccedil;ant par 0 <br />';	
 //	ALTER TABLE `gs_adherent` CHANGE `cp_adht` `cp_adht` VARCHAR( 8 ) NOT NULL DEFAULT '0'
 // POUR mySql
 //  ALTER TABLE "gsa_adherent" ALTER COLUMN "cp_adht" TYPE character varying(8)
 // Pour Postrgresql  integer -> character varying(8)
 //	Colonne		Type			
-// 	cp_adht		character varying(8
+// 	cp_adht		character varying(8)
 
-	//Si Mysql
-	if ( TYPE_BD_AODB == 'mysql' ) {
-		$req_upgrade_adht = "ALTER TABLE ".TABLE_ADHERENTS." CHANGE `cp_adht` `cp_adht` VARCHAR( 8 ) NOT NULL DEFAULT '0' ";
-		echo $req_upgrade_adht .' <br />';	
-		echo ' <br />';	
-		$dbresult = $db->Execute($req_upgrade_adht);	
-			if ($dbresult == true) {
-				echo 'Mysql - Modification r&eacute;ussie : champ "cp_adht" <br />';
-				$valid_modif = 1;
-			} else {
-				echo '<span class="erreur-Jaunerouge">Erreur requ&ecirc;te ou champ "cp_adht" ! </span><br />';
-				$valid_modif = 0;
-			}
-	}
-	
-	//Si postgresql
-	if ( TYPE_BD == 'postgres' ) {	
-		$req_upgrade_adht = "ALTER TABLE ".TABLE_ADHERENTS." ALTER COLUMN cp_adht TYPE character varying(8) ";
-		echo $req_upgrade_adht .' <br />';	
-		echo ' <br />';	
-		$dbresult = $db->Execute($req_upgrade_adht);	
-			if ($dbresult == true) {
-				echo 'PostgreSql - Modification r&eacute;ussie : champ "cp_adht" <br />';
-				$valid_modif = 1;
-			} else {
-				echo '<span class="erreur-Jaunerouge">Erreur requ&ecirc;te ou champ "cp_adht" ! </span><br />';
-				$valid_modif = 0;
-			}	
-	}
-
-		//echo ' <br />'.$valid_modif;			
-		if ($valid_modif == 1) {
-			echo ' Vous pouvez vous connectez &agrave; <a href="../index.php" title="Connexion">GestAssoPhp+Pg</a>, et v&eacute;rifier le fonctionnement dans Administration<br />';
+	$sqlarray = $dbdict->AlterColumnSQL(TABLE_ADHERENTS,'cp_adht C(8)`');
+	$return = $dbdict->ExecuteSQLArray($sqlarray);
+	//echo 	$return; // 2 Ok   1 errreur	
+		if ($return == 2) {
+		echo 'Modification r&eacute;ussie : champ "cp_adht" <br />';
 		} else {
-			echo '<span class="erreur-Jaunerouge">Erreur requ&ecirc;te - Contacter votre administrateur syst&egrave;me! </span><br />';
+		echo '<span class="erreur-Jaunerouge"><span class="erreur-Jaunerouge">Erreur requ&ecirc;te ou champ "cp_adht" ! </span><br />'.$db->ErrorMsg().'<br />';		
 		}
-
-
-
+	
+	echo '<br />------------------------------------------- <br />';	
+	echo '<b>2- Modifications Champs cp_adht dans table '.TABLE_COTISATIONS.'</b><br />Ajout colonne paiement_cotis pour une zone de PAIEMENT dans Gestion Cotisations<br />';	
+	// On upgrade TABLE_COTISATION  POUR mySql paiement_cotis	char(3)
+	// Pour Postrgresql character varying(3)
+	$sqlarray = $dbdict->AddColumnSQL(TABLE_COTISATIONS,'paiement_cotis C(3)`');
+	$return = $dbdict->ExecuteSQLArray($sqlarray);
+	//echo 	$return; // 2 Ok   1 errreur	
+		if ($return == 2) {
+		echo 'Modification r&eacute;ussie : ajout champ "paiement_cotis" <br />';
+		} else {
+		echo '<span class="erreur-Jaunerouge">Erreur requ&ecirc;te ou champ "paiement_cotis" d&eacute;j&agrave; existant ! </span><br />'.$db->ErrorMsg().'<br />';		
+		}	
+	
+	echo '<br />------------------------------------------- <br />';	
+	echo 'Vous pouvez retourner sur <a href="../admin/" title="Connexion">GestAssoPhp/Administration</a><br /><br />';	
+	echo '<span class="erreur-Jaunerouge">Supprimer le fichier admin/upgrade640-650.php pour des raison de s&eacute;curit&eacute;</span><br />';
+		
 
 echo '  
   </body>
diff --git a/gestion/gestassophp_sa/config/lang.php b/gestion/gestassophp_sa/config/lang.php
index 7e62734..59031d0 100644
--- a/gestion/gestassophp_sa/config/lang.php
+++ b/gestion/gestassophp_sa/config/lang.php
@@ -479,6 +479,7 @@ l'UTF-8, qui permet théoriquement d'encoder toutes les langues, du français au
 	define('_LANG_FICHE_COTIS_ADHT_DATE_DEB', 'Date d&eacute;but cotisation');
 	define('_LANG_FICHE_COTIS_ADHT_DATE_FIN', 'Date fin cotisation');
 	define('_LANG_FICHE_COTIS_ADHT_COMM', 'Commentaire');
+	define('_LANG_FICHE_COTIS_ADHT_MPAIE', 'Moyen de paiement'); //+ Ajout Zone PAIEMENT Gestion Cotisations
 	define('_LANG_FICHE_COTIS_ADHT_COMM_TITLE', 'Information sur la cotisation');//title=
 	define('_LANG_FICHE_COTIS_ADHT_RAISON', 'Raison de l\'archivage');
 	define('_LANG_FICHE_COTIS_ADHT_RAISON_TITLE', 'Information sur archivage');// title=
@@ -488,7 +489,7 @@ l'UTF-8, qui permet théoriquement d'encoder toutes les langues, du français au
 	// Add pour Visulisation avant impression  Ajout le 27/01/2009
 	define('_LANG_FICHE_COTIS_ADHT_VISU_BUTTON', 'Visualiser');	
 	define('_LANG_FICHE_COTIS_ADHT_VISU_BUTTON_TITLE', 'Pour impression &eacute;ventuelle');
-	define('_LANG_TITRE_ADMIN_FICHE_VISU_COTIS_ADHT', 'Reçu cotisation ');
+	define('_LANG_TITRE_ADMIN_FICHE_VISU_COTIS_ADHT', 'Re&ccedil;u cotisation ');
 	define('_LANG_FICHE_COTIS_ADHT_EURO', 'Euros');
 	define('_LANG_FICHE_COTIS_VISU_FAITLE', 'Fait le ');	
 	//+ ajout montant cotisation si cr&eacute;ation fiche
@@ -600,6 +601,9 @@ l'UTF-8, qui permet théoriquement d'encoder toutes les langues, du français au
 	$T_TRANCHE_AGE = array( ''=>'', 'Moins de 25 ans'=>'Moins de 25 ans','de 25 a 55 ans'=>'de 25 a 55 ans', 'Plus de 55 ans'=>'Plus de 55 ans');
 	$T_OUI_NON = array ('Non'=>'Non','Oui'=>'Oui',); 
 	// Attention  Only modify  the text after  =>   eg.  $T_OUI_NON = array ('Non'=>'No','Oui'=>'Yes', );
+	
+// Ajout Zone PAIEMENT Gestion Cotisations	 html_options name="paiement_cotis"  remplir_cotisations_adht
+	$T_PAIEMENT_COTIS = array('chq'=>'Ch&egrave;que','esp'=>'Esp&egrave;ces', 'c_b'=>'Carte bancaire', 'aut'=>'Autre');
 
 /** ---------------------------------------------------------------------- */	
 	
diff --git a/gestion/gestassophp_sa/doc/Changelog.txt b/gestion/gestassophp_sa/doc/Changelog.txt
index 605de72..7823ea7 100644
--- a/gestion/gestassophp_sa/doc/Changelog.txt
+++ b/gestion/gestassophp_sa/doc/Changelog.txt
@@ -12,12 +12,18 @@
  * @package   GestAssoPhp+Pg
  */
 
- 6.5.0 20/05/2013 en cours
+ 6.5.0 30/06/2013 en cours
  -----------------------
- - Modification en base de donnée du champ CP int(8) par VARCHAR( 8 ) pour  les Codes Postaux avec 0 comme 1er chiffre
+Attention modifications de base de données
 	Faites pointer votre navigateur sur l'URL d'installation (ex : http://votreserveur/gestion_s/gestassophp_s/admin/upgrade640-650.php) 
-	une page s'affiche ... "MODIFICATION BASE DE DONNEES " , vérifier si tous les messages sont OK, puis "cliquez sur le lien en bas de page
- - Vérifier sur votre serveur que les fonctionnalité 'magic_quotes_...' sont bien à OFF.
+	une page s'affiche ... "MODIFICATION BASE DE DONNEES " , vérifier si tous les messages sont OK, puis supprimer le fichier admin/upgrade640-650.php
+- Vérifier sur votre serveur que les fonctionnalité 'magic_quotes_...' sont bien à OFF.	
+- Ajout d'une zone Moyen de paiement sur Gestion Cotisations : Espèces(esp) Carte bancaire(c_b) Autre(aut)
+  Donc ajout d'une colonne paiement_cotis char(3) sur table xx_cotisations
+- Modification en base de donnée sur table xx_adherent du champ CP int(8) par VARCHAR( 8 ) pour les Codes Postaux avec 0 comme 1er chiffre
+- Mise à jour de Smarty 3.1.14 Jun 28, 2013
+
+ 
  
 6.4.0 05/04/2013 Finale
 -----------------------
diff --git a/gestion/gestassophp_sa/doc/tables_desciption.rtf b/gestion/gestassophp_sa/doc/tables_desciption.rtf
index c3c7be6..767f871 100644
--- a/gestion/gestassophp_sa/doc/tables_desciption.rtf
+++ b/gestion/gestassophp_sa/doc/tables_desciption.rtf

diff --git a/gestion/gestassophp_sa/install/schema.php b/gestion/gestassophp_sa/install/schema.php
index c13be38..2afac02 100644
--- a/gestion/gestassophp_sa/install/schema.php
+++ b/gestion/gestassophp_sa/install/schema.php
@@ -137,6 +137,7 @@ l'UTF-8, qui permet théoriquement d'encoder toutes les langues, du français au
 	   date_debut_cotis  D  NOT NULL ,
 	   date_fin_cotis  D  NOT NULL ,
 	   cotis  char(3) NOT NULL default '',
+	   paiement_cotis char(3) NOT NULL default '',
 	   info_archiv_cotis  C(30) NOT NULL default '',
 	   datemodiffiche_cotis  D  NOT NULL 
 	";
diff --git a/gestion/gestassophp_sa/templates/adherent/consulter_cotisations_adht.tpl b/gestion/gestassophp_sa/templates/adherent/consulter_cotisations_adht.tpl
index 5c5958e..6861f35 100644
--- a/gestion/gestassophp_sa/templates/adherent/consulter_cotisations_adht.tpl
+++ b/gestion/gestassophp_sa/templates/adherent/consulter_cotisations_adht.tpl
@@ -52,17 +52,18 @@
 		</tr>		
 		<tr>
 			<th class="LignegrisRight"> {language name=fiche_cotis_adht_date_fin} :</th>		
-			<td>{$data_cotis_adh.date_fin_cotis}</td>
+			<td>{$data_cotis_adh.date_fin_cotis} </td>
 		</tr>			
 		<tr>		
 			<td>&nbsp;</td><td>&nbsp;</td>
 		</tr>		
-		<tr>
-			<th class="LignegrisRight"> {language name=fiche_cotis_adht_comm} :</th>		
-			<td>{$data_cotis_adh.info_cotis}</td>
+		<tr><!-- Ajout Zone PAIEMENT Gestion Cotisations -->
+			<th class="LignegrisRight"> {language name=fiche_cotis_adht_mpaie} :</th>		
+			<td>{$data_cotis_adh.paiement_cotis}</td>
 		</tr>
 		<tr>		
-			<td>&nbsp;</td><td>&nbsp;</td>
+			<th class="LignegrisRight"> {language name=fiche_cotis_adht_comm} :</th>		
+			<td>{$data_cotis_adh.info_cotis}</td>
 		</tr>
 		<tr>		
 			<td>&nbsp;</td><td>&nbsp;</td>
diff --git a/gestion/gestassophp_sa/templates/adherent/remplir_cotisations_adht.tpl b/gestion/gestassophp_sa/templates/adherent/remplir_cotisations_adht.tpl
index 557f62b..2bf9bb1 100644
--- a/gestion/gestassophp_sa/templates/adherent/remplir_cotisations_adht.tpl
+++ b/gestion/gestassophp_sa/templates/adherent/remplir_cotisations_adht.tpl
@@ -79,6 +79,17 @@
 			<td><input type="text" name="date_fin_cotis" id="date_fin_cotis" title="{language name=tpl_texte_date_title}" value="{if $data_cotis_adh.date_fin_cotis}{$data_cotis_adh.date_fin_cotis}{else}{$date_3112}{/if}" size="12" maxlength="12" tabindex="6" {$disabled.date_fin_cotis} />{if $alert_saisie.d_fin_cotis_alert}<span class="erreur-Jaunerouge">&nbsp;{language name=message_cotis_adht_alert_archiv}</span>{/if} {if $erreur_saisie.d_fin_cotis}<span class="erreur-Jaunerouge">&nbsp;{$erreur_saisie.d_fin_cotis}</span>{/if}</td>
 		</tr>		
 		
+<!-- Ajout Zone PAIEMENT Gestion Cotisations -->
+		<tr>
+			<th class="LignegrisC"> {language name=fiche_cotis_adht_mpaie}</th>		
+			<td>
+			{if $archive_fiche == "1" || $supprime_fiche =="1"}{html_options name="paiement_cotis" options=$list_paiement_cotis selected=$data_cotis_adh.paiement_cotis tabindex="8" disabled="disabled"}
+			{else}
+			{html_options name="paiement_cotis" options=$list_paiement_cotis selected=$data_cotis_adh.paiement_cotis tabindex="8"}
+			{/if}
+			</td>
+		</tr>		
+<!-- /Ajout Zone PAIEMENT Gestion Cotisations -->	
 		
 		<tr>
 			<th class="LignegrisC"> {language name=fiche_cotis_adht_comm}</th>		
diff --git a/gestion/lib/smarty/Smarty.class.php b/gestion/lib/smarty/Smarty.class.php
index fb8de4b..bf86d56 100644
--- a/gestion/lib/smarty/Smarty.class.php
+++ b/gestion/lib/smarty/Smarty.class.php
@@ -2,7 +2,7 @@
 /**
  * Project:     Smarty: the PHP compiling template engine
  * File:        Smarty.class.php
- * SVN:         $Id: Smarty.class.php 4694 2013-01-13 21:13:14Z uwe.tews@googlemail.com $
+ * SVN:         $Id: Smarty.class.php 4742 2013-06-17 13:30:49Z Uwe.Tews@googlemail.com $
  *
  * This library is free software; you can redistribute it and/or
  * modify it under the terms of the GNU Lesser General Public
@@ -129,6 +129,11 @@ class Smarty extends Smarty_Internal_TemplateBase {
     const CACHING_LIFETIME_CURRENT = 1;
     const CACHING_LIFETIME_SAVED = 2;
     /**
+     * define constant for clearing cache files be saved expiration datees
+     */
+    const CLEAR_EXPIRED = -1; 
+
+    /**
      * define compile check modes
      */
     const COMPILECHECK_OFF = 0;
@@ -1323,9 +1328,9 @@ class Smarty extends Smarty_Internal_TemplateBase {
      * @param int $max_errors
      * @return integer number of template files recompiled
      */
-    public function compileAllTemplates($extention = '.tpl', $force_compile = false, $time_limit = 0, $max_errors = null)
+    public function compileAllTemplates($extension = '.tpl', $force_compile = false, $time_limit = 0, $max_errors = null)
     {
-        return Smarty_Internal_Utility::compileAllTemplates($extention, $force_compile, $time_limit, $max_errors, $this);
+        return Smarty_Internal_Utility::compileAllTemplates($extension, $force_compile, $time_limit, $max_errors, $this);
     }
 
     /**
@@ -1337,9 +1342,9 @@ class Smarty extends Smarty_Internal_TemplateBase {
      * @param int $max_errors
      * @return integer number of template files recompiled
      */
-    public function compileAllConfig($extention = '.conf', $force_compile = false, $time_limit = 0, $max_errors = null)
+    public function compileAllConfig($extension = '.conf', $force_compile = false, $time_limit = 0, $max_errors = null)
     {
-        return Smarty_Internal_Utility::compileAllConfig($extention, $force_compile, $time_limit, $max_errors, $this);
+        return Smarty_Internal_Utility::compileAllConfig($extension, $force_compile, $time_limit, $max_errors, $this);
     }
 
     /**
@@ -1525,4 +1530,4 @@ function smartyAutoload($class)
     }
 }
 
-?>
+?>
\ No newline at end of file
diff --git a/gestion/lib/smarty/change_log.txt b/gestion/lib/smarty/change_log.txt
index 8527ab4..e98f519 100644
--- a/gestion/lib/smarty/change_log.txt
+++ b/gestion/lib/smarty/change_log.txt
@@ -1,4 +1,40 @@
-﻿===== trunk  =====
+﻿
+
+===== Smarty-3.1.14 =====
+27.06.2013
+- bugfix removed PHP 5.5 deprecated preg_replace /e option in modifier capitalize (forum topic 24389)
+
+17.06.2013
+- fixed spelling in sources and documentation (from smarty-developers forum Veres Lajos)
+- enhancement added constant SMARTY::CLEAR_EXPIRED for the change of 26.05.2013 (forum topic 24310)
+- bugfix added smarty_security.php to composer.json (Issue 135)
+
+26.05.2013
+- enhancement an expire_time of -1 in clearCache() and clearAllCache() will delete outdated cache files
+by their individual cache_lifetime used at creation (forum topic 24310)
+
+21.05.2013
+- bugfix modifier strip_tags:true was compiled into wrong code (Forum Topic 24287)
+- bugfix /n after ?> in Smarty.class.php did start output buffering (Issue 138)
+
+25.04.2013
+- bugfix escape and wordrap modifier could be compiled into wrong code when used in {nocache}{/nocache}
+section but caching is disabled (Forum Topic 24260)
+
+05.04.2013
+- bugfix post filter must not run when compiling inheritance child blocks (Forum Topic 24094)
+- bugfix after the fix for Issue #130 compiler exceptions got double escaped (Forum Topic 24199)
+
+28.02.2013
+- bugfix nocache blocks could be lost when using CACHING_LIFETIME_SAVED (Issue #133)
+- bugfix Compile ID gets nulled when compiling child blocks (Issue #134)
+
+
+24.01.2013
+- bugfix wrong tag type in smarty_internal_templatecompilerbase.php could cause wrong plugin search order (Forum Topic 24028)
+
+
+===== Smarty-3.1.13 =====
 13.01.2013
 - enhancement allow to disable exception message escaping by SmartyException::$escape = false;  (Issue #130)
 
diff --git a/gestion/lib/smarty/sysplugins/smarty_cacheresource.php b/gestion/lib/smarty/sysplugins/smarty_cacheresource.php
index ca18add..a685299 100644
--- a/gestion/lib/smarty/sysplugins/smarty_cacheresource.php
+++ b/gestion/lib/smarty/sysplugins/smarty_cacheresource.php
@@ -224,7 +224,7 @@ class Smarty_Template_Cached {
     public $timestamp = false;
 
     /**
-    * Source Existance
+    * Source Existence
     * @var boolean
     */
     public $exists = false;
diff --git a/gestion/lib/smarty/sysplugins/smarty_internal_cacheresource_file.php b/gestion/lib/smarty/sysplugins/smarty_internal_cacheresource_file.php
index 04194b3..fca897d 100644
--- a/gestion/lib/smarty/sysplugins/smarty_internal_cacheresource_file.php
+++ b/gestion/lib/smarty/sysplugins/smarty_internal_cacheresource_file.php
@@ -1,266 +1,275 @@
 <?php
-/**
- * Smarty Internal Plugin CacheResource File
- *
- * @package Smarty
- * @subpackage Cacher
- * @author Uwe Tews
- * @author Rodney Rehm
- */
-
-/**
- * This class does contain all necessary methods for the HTML cache on file system
- *
- * Implements the file system as resource for the HTML cache Version ussing nocache inserts.
- *
- * @package Smarty
- * @subpackage Cacher
- */
-class Smarty_Internal_CacheResource_File extends Smarty_CacheResource {
+    /**
+    * Smarty Internal Plugin CacheResource File
+    *
+    * @package Smarty
+    * @subpackage Cacher
+    * @author Uwe Tews
+    * @author Rodney Rehm
+    */
 
     /**
-     * populate Cached Object with meta data from Resource
-     *
-     * @param Smarty_Template_Cached   $cached    cached object
-     * @param Smarty_Internal_Template $_template template object
-     * @return void
-     */
-    public function populate(Smarty_Template_Cached $cached, Smarty_Internal_Template $_template)
-    {
-        $_source_file_path = str_replace(':', '.', $_template->source->filepath);
-        $_cache_id = isset($_template->cache_id) ? preg_replace('![^\w\|]+!', '_', $_template->cache_id) : null;
-        $_compile_id = isset($_template->compile_id) ? preg_replace('![^\w\|]+!', '_', $_template->compile_id) : null;
-        $_filepath = $_template->source->uid;
-        // if use_sub_dirs, break file into directories
-        if ($_template->smarty->use_sub_dirs) {
-            $_filepath = substr($_filepath, 0, 2) . DS
+    * This class does contain all necessary methods for the HTML cache on file system
+    *
+    * Implements the file system as resource for the HTML cache Version ussing nocache inserts.
+    *
+    * @package Smarty
+    * @subpackage Cacher
+    */
+    class Smarty_Internal_CacheResource_File extends Smarty_CacheResource {
+
+        /**
+        * populate Cached Object with meta data from Resource
+        *
+        * @param Smarty_Template_Cached   $cached    cached object
+        * @param Smarty_Internal_Template $_template template object
+        * @return void
+        */
+        public function populate(Smarty_Template_Cached $cached, Smarty_Internal_Template $_template)
+        {
+            $_source_file_path = str_replace(':', '.', $_template->source->filepath);
+            $_cache_id = isset($_template->cache_id) ? preg_replace('![^\w\|]+!', '_', $_template->cache_id) : null;
+            $_compile_id = isset($_template->compile_id) ? preg_replace('![^\w\|]+!', '_', $_template->compile_id) : null;
+            $_filepath = $_template->source->uid;
+            // if use_sub_dirs, break file into directories
+            if ($_template->smarty->use_sub_dirs) {
+                $_filepath = substr($_filepath, 0, 2) . DS
                 . substr($_filepath, 2, 2) . DS
                 . substr($_filepath, 4, 2) . DS
                 . $_filepath;
-        }
-        $_compile_dir_sep = $_template->smarty->use_sub_dirs ? DS : '^';
-        if (isset($_cache_id)) {
-            $_cache_id = str_replace('|', $_compile_dir_sep, $_cache_id) . $_compile_dir_sep;
-        } else {
-            $_cache_id = '';
-        }
-        if (isset($_compile_id)) {
-            $_compile_id = $_compile_id . $_compile_dir_sep;
-        } else {
-            $_compile_id = '';
-        }
-        $_cache_dir = $_template->smarty->getCacheDir();
-        if ($_template->smarty->cache_locking) {
-            // create locking file name
-            // relative file name?
-            if (!preg_match('/^([\/\\\\]|[a-zA-Z]:[\/\\\\])/', $_cache_dir)) {
-                $_lock_dir = rtrim(getcwd(), '/\\') . DS . $_cache_dir;
+            }
+            $_compile_dir_sep = $_template->smarty->use_sub_dirs ? DS : '^';
+            if (isset($_cache_id)) {
+                $_cache_id = str_replace('|', $_compile_dir_sep, $_cache_id) . $_compile_dir_sep;
+            } else {
+                $_cache_id = '';
+            }
+            if (isset($_compile_id)) {
+                $_compile_id = $_compile_id . $_compile_dir_sep;
             } else {
-                $_lock_dir = $_cache_dir;
+                $_compile_id = '';
             }
-            $cached->lock_id = $_lock_dir.sha1($_cache_id.$_compile_id.$_template->source->uid).'.lock';
+            $_cache_dir = $_template->smarty->getCacheDir();
+            if ($_template->smarty->cache_locking) {
+                // create locking file name
+                // relative file name?
+                if (!preg_match('/^([\/\\\\]|[a-zA-Z]:[\/\\\\])/', $_cache_dir)) {
+                    $_lock_dir = rtrim(getcwd(), '/\\') . DS . $_cache_dir;
+                } else {
+                    $_lock_dir = $_cache_dir;
+                }
+                $cached->lock_id = $_lock_dir.sha1($_cache_id.$_compile_id.$_template->source->uid).'.lock';
+            }
+            $cached->filepath = $_cache_dir . $_cache_id . $_compile_id . $_filepath . '.' . basename($_source_file_path) . '.php';
+            $cached->timestamp = @filemtime($cached->filepath);
+            $cached->exists = !!$cached->timestamp;
         }
-        $cached->filepath = $_cache_dir . $_cache_id . $_compile_id . $_filepath . '.' . basename($_source_file_path) . '.php';
-        $cached->timestamp = @filemtime($cached->filepath);
-        $cached->exists = !!$cached->timestamp;
-    }
 
-    /**
-     * populate Cached Object with timestamp and exists from Resource
-     *
-     * @param Smarty_Template_Cached $cached cached object
-     * @return void
-     */
-    public function populateTimestamp(Smarty_Template_Cached $cached)
-    {
-        $cached->timestamp = @filemtime($cached->filepath);
-        $cached->exists = !!$cached->timestamp;
-    }
+        /**
+        * populate Cached Object with timestamp and exists from Resource
+        *
+        * @param Smarty_Template_Cached $cached cached object
+        * @return void
+        */
+        public function populateTimestamp(Smarty_Template_Cached $cached)
+        {
+            $cached->timestamp = @filemtime($cached->filepath);
+            $cached->exists = !!$cached->timestamp;
+        }
 
-    /**
-     * Read the cached template and process its header
-     *
-     * @param Smarty_Internal_Template $_template template object
-     * @param Smarty_Template_Cached $cached cached object
-     * @return booelan true or false if the cached content does not exist
-     */
-    public function process(Smarty_Internal_Template $_template, Smarty_Template_Cached $cached=null)
-    {
-        $_smarty_tpl = $_template;
-        return @include $_template->cached->filepath;
-    }
+        /**
+        * Read the cached template and process its header
+        *
+        * @param Smarty_Internal_Template $_template template object
+        * @param Smarty_Template_Cached $cached cached object
+        * @return booelan true or false if the cached content does not exist
+        */
+        public function process(Smarty_Internal_Template $_template, Smarty_Template_Cached $cached=null)
+        {
+            $_smarty_tpl = $_template;
+            return @include $_template->cached->filepath;
+        }
 
-    /**
-     * Write the rendered template output to cache
-     *
-     * @param Smarty_Internal_Template $_template template object
-     * @param string                   $content   content to cache
-     * @return boolean success
-     */
-    public function writeCachedContent(Smarty_Internal_Template $_template, $content)
-    {
-        if (Smarty_Internal_Write_File::writeFile($_template->cached->filepath, $content, $_template->smarty) === true) {
-            $_template->cached->timestamp = @filemtime($_template->cached->filepath);
-            $_template->cached->exists = !!$_template->cached->timestamp;
-            if ($_template->cached->exists) {
-                return true;
+        /**
+        * Write the rendered template output to cache
+        *
+        * @param Smarty_Internal_Template $_template template object
+        * @param string                   $content   content to cache
+        * @return boolean success
+        */
+        public function writeCachedContent(Smarty_Internal_Template $_template, $content)
+        {
+            if (Smarty_Internal_Write_File::writeFile($_template->cached->filepath, $content, $_template->smarty) === true) {
+                $_template->cached->timestamp = @filemtime($_template->cached->filepath);
+                $_template->cached->exists = !!$_template->cached->timestamp;
+                if ($_template->cached->exists) {
+                    return true;
+                }
             }
+            return false;
         }
-        return false;
-    }
 
-    /**
-     * Empty cache
-     *
-     * @param Smarty_Internal_Template $_template template object
-     * @param integer                  $exp_time  expiration time (number of seconds, not timestamp)
-     * @return integer number of cache files deleted
-     */
-    public function clearAll(Smarty $smarty, $exp_time = null)
-    {
-        return $this->clear($smarty, null, null, null, $exp_time);
-    }
+        /**
+        * Empty cache
+        *
+        * @param Smarty_Internal_Template $_template template object
+        * @param integer                  $exp_time  expiration time (number of seconds, not timestamp)
+        * @return integer number of cache files deleted
+        */
+        public function clearAll(Smarty $smarty, $exp_time = null)
+        {
+            return $this->clear($smarty, null, null, null, $exp_time);
+        }
 
-    /**
-     * Empty cache for a specific template
-     *
-     * @param Smarty  $_template     template object
-     * @param string  $resource_name template name
-     * @param string  $cache_id      cache id
-     * @param string  $compile_id    compile id
-     * @param integer $exp_time      expiration time (number of seconds, not timestamp)
-     * @return integer number of cache files deleted
-    */
-    public function clear(Smarty $smarty, $resource_name, $cache_id, $compile_id, $exp_time)
-    {
-        $_cache_id = isset($cache_id) ? preg_replace('![^\w\|]+!', '_', $cache_id) : null;
-        $_compile_id = isset($compile_id) ? preg_replace('![^\w\|]+!', '_', $compile_id) : null;
-        $_dir_sep = $smarty->use_sub_dirs ? '/' : '^';
-        $_compile_id_offset = $smarty->use_sub_dirs ? 3 : 0;
-        $_dir = $smarty->getCacheDir();
-        $_dir_length = strlen($_dir);
-        if (isset($_cache_id)) {
-            $_cache_id_parts = explode('|', $_cache_id);
-            $_cache_id_parts_count = count($_cache_id_parts);
-            if ($smarty->use_sub_dirs) {
-                foreach ($_cache_id_parts as $id_part) {
-                    $_dir .= $id_part . DS;
+        /**
+        * Empty cache for a specific template
+        *
+        * @param Smarty  $_template     template object
+        * @param string  $resource_name template name
+        * @param string  $cache_id      cache id
+        * @param string  $compile_id    compile id
+        * @param integer $exp_time      expiration time (number of seconds, not timestamp)
+        * @return integer number of cache files deleted
+        */
+        public function clear(Smarty $smarty, $resource_name, $cache_id, $compile_id, $exp_time)
+        {
+            $_cache_id = isset($cache_id) ? preg_replace('![^\w\|]+!', '_', $cache_id) : null;
+            $_compile_id = isset($compile_id) ? preg_replace('![^\w\|]+!', '_', $compile_id) : null;
+            $_dir_sep = $smarty->use_sub_dirs ? '/' : '^';
+            $_compile_id_offset = $smarty->use_sub_dirs ? 3 : 0;
+            $_dir = $smarty->getCacheDir();
+            $_dir_length = strlen($_dir);
+            if (isset($_cache_id)) {
+                $_cache_id_parts = explode('|', $_cache_id);
+                $_cache_id_parts_count = count($_cache_id_parts);
+                if ($smarty->use_sub_dirs) {
+                    foreach ($_cache_id_parts as $id_part) {
+                        $_dir .= $id_part . DS;
+                    }
                 }
             }
-        }
-        if (isset($resource_name)) {
-            $_save_stat = $smarty->caching;
-            $smarty->caching = true;
-            $tpl = new $smarty->template_class($resource_name, $smarty);
-            $smarty->caching = $_save_stat;
+            if (isset($resource_name)) {
+                $_save_stat = $smarty->caching;
+                $smarty->caching = true;
+                $tpl = new $smarty->template_class($resource_name, $smarty);
+                $smarty->caching = $_save_stat;
 
-            // remove from template cache
-            $tpl->source; // have the template registered before unset()
-            if ($smarty->allow_ambiguous_resources) {
-                $_templateId = $tpl->source->unique_resource . $tpl->cache_id . $tpl->compile_id;
-            } else {
-                $_templateId = $smarty->joined_template_dir . '#' . $resource_name . $tpl->cache_id . $tpl->compile_id;
-            }
-            if (isset($_templateId[150])) {
-                $_templateId = sha1($_templateId);
-            }
-            unset($smarty->template_objects[$_templateId]);
+                // remove from template cache
+                $tpl->source; // have the template registered before unset()
+                if ($smarty->allow_ambiguous_resources) {
+                    $_templateId = $tpl->source->unique_resource . $tpl->cache_id . $tpl->compile_id;
+                } else {
+                    $_templateId = $smarty->joined_template_dir . '#' . $resource_name . $tpl->cache_id . $tpl->compile_id;
+                }
+                if (isset($_templateId[150])) {
+                    $_templateId = sha1($_templateId);
+                }
+                unset($smarty->template_objects[$_templateId]);
 
-            if ($tpl->source->exists) {
-                $_resourcename_parts = basename(str_replace('^', '/', $tpl->cached->filepath));
-            } else {
-                return 0;
-            }
-        }
-        $_count = 0;
-        $_time = time();
-        if (file_exists($_dir)) {
-            $_cacheDirs = new RecursiveDirectoryIterator($_dir);
-            $_cache = new RecursiveIteratorIterator($_cacheDirs, RecursiveIteratorIterator::CHILD_FIRST);
-            foreach ($_cache as $_file) {
-                if (substr(basename($_file->getPathname()),0,1) == '.' || strpos($_file, '.svn') !== false) continue;
-                // directory ?
-                if ($_file->isDir()) {
-                    if (!$_cache->isDot()) {
-                        // delete folder if empty
-                        @rmdir($_file->getPathname());
-                    }
+                if ($tpl->source->exists) {
+                    $_resourcename_parts = basename(str_replace('^', '/', $tpl->cached->filepath));
                 } else {
-                    $_parts = explode($_dir_sep, str_replace('\\', '/', substr((string)$_file, $_dir_length)));
-                    $_parts_count = count($_parts);
-                    // check name
-                    if (isset($resource_name)) {
-                        if ($_parts[$_parts_count-1] != $_resourcename_parts) {
-                            continue;
+                    return 0;
+                }
+            }
+            $_count = 0;
+            $_time = time();
+            if (file_exists($_dir)) {
+                $_cacheDirs = new RecursiveDirectoryIterator($_dir);
+                $_cache = new RecursiveIteratorIterator($_cacheDirs, RecursiveIteratorIterator::CHILD_FIRST);
+                foreach ($_cache as $_file) {
+                    if (substr(basename($_file->getPathname()),0,1) == '.' || strpos($_file, '.svn') !== false) continue;
+                    // directory ?
+                    if ($_file->isDir()) {
+                        if (!$_cache->isDot()) {
+                            // delete folder if empty
+                            @rmdir($_file->getPathname());
                         }
-                    }
-                    // check compile id
-                    if (isset($_compile_id) && (!isset($_parts[$_parts_count-2 - $_compile_id_offset]) || $_parts[$_parts_count-2 - $_compile_id_offset] != $_compile_id)) {
-                        continue;
-                    }
-                    // check cache id
-                    if (isset($_cache_id)) {
-                        // count of cache id parts
-                        $_parts_count = (isset($_compile_id)) ? $_parts_count - 2 - $_compile_id_offset : $_parts_count - 1 - $_compile_id_offset;
-                        if ($_parts_count < $_cache_id_parts_count) {
+                    } else {
+                        $_parts = explode($_dir_sep, str_replace('\\', '/', substr((string)$_file, $_dir_length)));
+                        $_parts_count = count($_parts);
+                        // check name
+                        if (isset($resource_name)) {
+                            if ($_parts[$_parts_count-1] != $_resourcename_parts) {
+                                continue;
+                            }
+                        }
+                        // check compile id
+                        if (isset($_compile_id) && (!isset($_parts[$_parts_count-2 - $_compile_id_offset]) || $_parts[$_parts_count-2 - $_compile_id_offset] != $_compile_id)) {
                             continue;
                         }
-                        for ($i = 0; $i < $_cache_id_parts_count; $i++) {
-                            if ($_parts[$i] != $_cache_id_parts[$i]) continue 2;
+                        // check cache id
+                        if (isset($_cache_id)) {
+                            // count of cache id parts
+                            $_parts_count = (isset($_compile_id)) ? $_parts_count - 2 - $_compile_id_offset : $_parts_count - 1 - $_compile_id_offset;
+                            if ($_parts_count < $_cache_id_parts_count) {
+                                continue;
+                            }
+                            for ($i = 0; $i < $_cache_id_parts_count; $i++) {
+                                if ($_parts[$i] != $_cache_id_parts[$i]) continue 2;
+                            }
                         }
+                        // expired ? 
+                        if  (isset($exp_time)) {
+                            if ($exp_time < 0) {
+                                preg_match('#\'cache_lifetime\' =>\s*(\d*)#', file_get_contents($_file), $match); 
+                                if ($_time < (@filemtime($_file) + $match[1])) { 
+                                    continue; 
+                                }
+                            } else {
+                                if ($_time - @filemtime($_file) < $exp_time) {
+                                    continue;
+                                } 
+                            }
+                        }
+                        $_count += @unlink((string) $_file) ? 1 : 0;
                     }
-                    // expired ?
-                    if (isset($exp_time) && $_time - @filemtime($_file) < $exp_time) {
-                        continue;
-                    }
-                    $_count += @unlink((string) $_file) ? 1 : 0;
                 }
             }
+            return $_count;
         }
-        return $_count;
-    }
 
-    /**
-     * Check is cache is locked for this template
-     *
-     * @param Smarty $smarty Smarty object
-     * @param Smarty_Template_Cached $cached cached object
-     * @return booelan true or false if cache is locked
-     */
-    public function hasLock(Smarty $smarty, Smarty_Template_Cached $cached)
-    {
-        if (version_compare(PHP_VERSION, '5.3.0', '>=')) {
-            clearstatcache(true, $cached->lock_id);
-        } else {
-            clearstatcache();
+        /**
+        * Check is cache is locked for this template
+        *
+        * @param Smarty $smarty Smarty object
+        * @param Smarty_Template_Cached $cached cached object
+        * @return booelan true or false if cache is locked
+        */
+        public function hasLock(Smarty $smarty, Smarty_Template_Cached $cached)
+        {
+            if (version_compare(PHP_VERSION, '5.3.0', '>=')) {
+                clearstatcache(true, $cached->lock_id);
+            } else {
+                clearstatcache();
+            }
+            $t = @filemtime($cached->lock_id);
+            return $t && (time() - $t < $smarty->locking_timeout);
         }
-        $t = @filemtime($cached->lock_id);
-        return $t && (time() - $t < $smarty->locking_timeout);
-    }
 
-    /**
-     * Lock cache for this template
-     *
-     * @param Smarty $smarty Smarty object
-     * @param Smarty_Template_Cached $cached cached object
-     */
-    public function acquireLock(Smarty $smarty, Smarty_Template_Cached $cached)
-    {
-        $cached->is_locked = true;
-        touch($cached->lock_id);
-    }
+        /**
+        * Lock cache for this template
+        *
+        * @param Smarty $smarty Smarty object
+        * @param Smarty_Template_Cached $cached cached object
+        */
+        public function acquireLock(Smarty $smarty, Smarty_Template_Cached $cached)
+        {
+            $cached->is_locked = true;
+            touch($cached->lock_id);
+        }
 
-    /**
-     * Unlock cache for this template
-     *
-     * @param Smarty $smarty Smarty object
-     * @param Smarty_Template_Cached $cached cached object
-     */
-    public function releaseLock(Smarty $smarty, Smarty_Template_Cached $cached)
-    {
-        $cached->is_locked = false;
-        @unlink($cached->lock_id);
+        /**
+        * Unlock cache for this template
+        *
+        * @param Smarty $smarty Smarty object
+        * @param Smarty_Template_Cached $cached cached object
+        */
+        public function releaseLock(Smarty $smarty, Smarty_Template_Cached $cached)
+        {
+            $cached->is_locked = false;
+            @unlink($cached->lock_id);
+        }
     }
-}
 
 ?>
\ No newline at end of file
diff --git a/gestion/lib/smarty/sysplugins/smarty_internal_compile_block.php b/gestion/lib/smarty/sysplugins/smarty_internal_compile_block.php
index f760e55..847b04a 100644
--- a/gestion/lib/smarty/sysplugins/smarty_internal_compile_block.php
+++ b/gestion/lib/smarty/sysplugins/smarty_internal_compile_block.php
@@ -84,7 +84,7 @@ class Smarty_Internal_Compile_Block extends Smarty_Internal_CompileBase {
             $al = '';
         }
         if (0 == preg_match("!({$_ldl}{$al}block\s+)(name=)?(\w+|'.*'|\".*\")(\s*?)?((append|prepend|nocache)?(\s*)?(hide)?)?(\s*{$_rdl})!", $block_tag, $_match)) {
-            $error_text = 'Syntax Error in template "' . $template->source->filepath . '"   "' . htmlspecialchars($block_tag) . '" illegal options';
+            $error_text = 'Syntax Error in template "' . $template->source->filepath . '"   "' . $block_tag . '" illegal options';
             throw new SmartyCompilerException($error_text);
         } else {
             $_name = trim($_match[3], '\'"');
@@ -179,7 +179,7 @@ class Smarty_Internal_Compile_Block extends Smarty_Internal_CompileBase {
             return '';
         }
         $_tpl = new Smarty_Internal_template('string:' . $compiler->template->block_data[$_name]['source'], $compiler->smarty, $compiler->template, $compiler->template->cache_id,
-                        $compiler->template->compile_id = null, $compiler->template->caching, $compiler->template->cache_lifetime);
+                        $compiler->template->compile_id, $compiler->template->caching, $compiler->template->cache_lifetime);
         $_tpl->variable_filters = $compiler->template->variable_filters;
         $_tpl->properties['nocache_hash'] = $compiler->template->properties['nocache_hash'];
         $_tpl->source->filepath = $compiler->template->block_data[$_name]['file'];
@@ -190,6 +190,7 @@ class Smarty_Internal_Compile_Block extends Smarty_Internal_CompileBase {
             $_tpl->compiler->forceNocache = 1;
         }
         $_tpl->compiler->suppressHeader = true;
+        $_tpl->compiler->suppressFilter = true;
         $_tpl->compiler->suppressTemplatePropertyHeader = true;
         $_tpl->compiler->suppressMergedTemplates = true;
         if (strpos($compiler->template->block_data[$_name]['source'], '%%%%SMARTY_PARENT%%%%') !== false) {
diff --git a/gestion/lib/smarty/sysplugins/smarty_internal_template.php b/gestion/lib/smarty/sysplugins/smarty_internal_template.php
index 29017ed..b6dd87f 100644
--- a/gestion/lib/smarty/sysplugins/smarty_internal_template.php
+++ b/gestion/lib/smarty/sysplugins/smarty_internal_template.php
@@ -467,10 +467,15 @@ class Smarty_Internal_Template extends Smarty_Internal_TemplateBase {
             }
         }
         if ($cache) {
+            // CACHING_LIFETIME_SAVED cache expiry has to be validated here since otherwise we'd define the unifunc
+            if ($this->caching === Smarty::CACHING_LIFETIME_SAVED &&
+                $this->properties['cache_lifetime'] >= 0 &&
+                (time() > ($this->cached->timestamp + $this->properties['cache_lifetime']))) {
+                $is_valid = false;
+            }
             $this->cached->valid = $is_valid;
         } else {
-            $this->mustCompile = !$is_valid;
-        }
+            $this->mustCompile = !$is_valid;        }
         // store data in reusable Smarty_Template_Compiled
         if (!$cache) {
             $this->compiled->_properties = $properties;
diff --git a/gestion/lib/smarty/sysplugins/smarty_internal_templatecompilerbase.php b/gestion/lib/smarty/sysplugins/smarty_internal_templatecompilerbase.php
index c745d29..e4a78d0 100644
--- a/gestion/lib/smarty/sysplugins/smarty_internal_templatecompilerbase.php
+++ b/gestion/lib/smarty/sysplugins/smarty_internal_templatecompilerbase.php
@@ -107,6 +107,12 @@ abstract class Smarty_Internal_TemplateCompilerBase {
     public $suppressTemplatePropertyHeader = false;
 
     /**
+     * suppress pre and post filter
+     * @var bool
+     */
+    public $suppressFilter = false;
+
+    /**
      * flag if compiled template file shall we written
      * @var bool
      */
@@ -184,7 +190,7 @@ abstract class Smarty_Internal_TemplateCompilerBase {
             // get template source
             $_content = $template->source->content;
             // run prefilter if required
-            if (isset($this->smarty->autoload_filters['pre']) || isset($this->smarty->registered_filters['pre'])) {
+            if ((isset($this->smarty->autoload_filters['pre']) || isset($this->smarty->registered_filters['pre'])) && !$this->suppressFilter) {
                 $_content = Smarty_Internal_Filter_Handler::runFilter('pre', $_content, $template);
             }
             // on empty template just return header
@@ -209,13 +215,9 @@ abstract class Smarty_Internal_TemplateCompilerBase {
             foreach ($this->merged_templates as $code) {
                 $merged_code .= $code;
             }
-            // run postfilter if required on merged code
-            if (isset($this->smarty->autoload_filters['post']) || isset($this->smarty->registered_filters['post'])) {
-                $merged_code = Smarty_Internal_Filter_Handler::runFilter('post', $merged_code, $template);
-            }
         }
         // run postfilter if required on compiled template code
-        if (isset($this->smarty->autoload_filters['post']) || isset($this->smarty->registered_filters['post'])) {
+        if ((isset($this->smarty->autoload_filters['post']) || isset($this->smarty->registered_filters['post'])) && !$this->suppressFilter) {
             $_compiled_code = Smarty_Internal_Filter_Handler::runFilter('post', $_compiled_code, $template);
         }
         if ($this->suppressTemplatePropertyHeader) {
@@ -331,7 +333,7 @@ abstract class Smarty_Internal_TemplateCompilerBase {
                 }
                 // check plugins from plugins folder
                 foreach ($this->smarty->plugin_search_order as $plugin_type) {
-                    if ($plugin_type == Smarty::PLUGIN_BLOCK && $this->smarty->loadPlugin('smarty_compiler_' . $tag) && (!isset($this->smarty->security_policy) || $this->smarty->security_policy->isTrustedTag($tag, $this))) {
+                    if ($plugin_type == Smarty::PLUGIN_COMPILER && $this->smarty->loadPlugin('smarty_compiler_' . $tag) && (!isset($this->smarty->security_policy) || $this->smarty->security_policy->isTrustedTag($tag, $this))) {
                         $plugin = 'smarty_compiler_' . $tag;
                         if (is_callable($plugin)) {
                             // convert arguments format for old compiler plugins
@@ -605,7 +607,7 @@ abstract class Smarty_Internal_TemplateCompilerBase {
                 $_output = addcslashes($content,'\'\\');
                 $_output = str_replace("^#^", "'", $_output);
                 $_output = "<?php echo '/*%%SmartyNocache:{$this->nocache_hash}%%*/" . $_output . "/*/%%SmartyNocache:{$this->nocache_hash}%%*/';?>\n";
-                // make sure we include modifer plugins for nocache code
+                // make sure we include modifier plugins for nocache code
                 foreach ($this->modifier_plugins as $plugin_name => $dummy) {
                     if (isset($this->template->required_plugins['compiled'][$plugin_name]['modifier'])) {
                         $this->template->required_plugins['nocache'][$plugin_name]['modifier'] = $this->template->required_plugins['compiled'][$plugin_name]['modifier'];
@@ -641,7 +643,7 @@ abstract class Smarty_Internal_TemplateCompilerBase {
             $line = $this->lex->line;
         }
         $match = preg_split("/\n/", $this->lex->data);
-        $error_text = 'Syntax Error in template "' . $this->template->source->filepath . '"  on line ' . $line . ' "' . htmlspecialchars(trim(preg_replace('![\t\r\n]+!', ' ', $match[$line - 1]))) . '" ';
+        $error_text = 'Syntax Error in template "' . $this->template->source->filepath . '"  on line ' . $line . ' "' . trim(preg_replace('![\t\r\n]+!', ' ', $match[$line - 1])) . '" ';
         if (isset($args)) {
             // individual error message
             $error_text .= $args;
diff --git a/gestion/lib/smarty/sysplugins/smarty_internal_utility.php b/gestion/lib/smarty/sysplugins/smarty_internal_utility.php
index f19ca01..0bec8db 100644
--- a/gestion/lib/smarty/sysplugins/smarty_internal_utility.php
+++ b/gestion/lib/smarty/sysplugins/smarty_internal_utility.php
@@ -58,7 +58,7 @@ class Smarty_Internal_Utility {
      * @param Smarty $smarty        Smarty instance
      * @return integer number of template files compiled
      */
-    public static function compileAllTemplates($extention, $force_compile, $time_limit, $max_errors, Smarty $smarty)
+    public static function compileAllTemplates($extension, $force_compile, $time_limit, $max_errors, Smarty $smarty)
     {
         // switch off time limit
         if (function_exists('set_time_limit')) {
@@ -74,7 +74,7 @@ class Smarty_Internal_Utility {
             foreach ($_compile as $_fileinfo) {
                 $_file = $_fileinfo->getFilename();
                 if (substr(basename($_fileinfo->getPathname()),0,1) == '.' || strpos($_file, '.svn') !== false) continue;
-                if (!substr_compare($_file, $extention, - strlen($extention)) == 0) continue;
+                if (!substr_compare($_file, $extension, - strlen($extension)) == 0) continue;
                 if ($_fileinfo->getPath() == substr($_dir, 0, -1)) {
                    $_template_file = $_file;
                 } else {
@@ -122,7 +122,7 @@ class Smarty_Internal_Utility {
      * @param Smarty $smarty        Smarty instance
      * @return integer number of config files compiled
      */
-    public static function compileAllConfig($extention, $force_compile, $time_limit, $max_errors, Smarty $smarty)
+    public static function compileAllConfig($extension, $force_compile, $time_limit, $max_errors, Smarty $smarty)
     {
         // switch off time limit
         if (function_exists('set_time_limit')) {
@@ -138,7 +138,7 @@ class Smarty_Internal_Utility {
             foreach ($_compile as $_fileinfo) {
                 $_file = $_fileinfo->getFilename();
                 if (substr(basename($_fileinfo->getPathname()),0,1) == '.' || strpos($_file, '.svn') !== false) continue;
-                if (!substr_compare($_file, $extention, - strlen($extention)) == 0) continue;
+                if (!substr_compare($_file, $extension, - strlen($extension)) == 0) continue;
                 if ($_fileinfo->getPath() == substr($_dir, 0, -1)) {
                     $_config_file = $_file;
                 } else {
@@ -308,7 +308,7 @@ class Smarty_Internal_Utility {
         foreach($smarty->getTemplateDir() as $template_dir) {
             $_template_dir = $template_dir;
             $template_dir = realpath($template_dir);
-            // resolve include_path or fail existance
+            // resolve include_path or fail existence
             if (!$template_dir) {
                 if ($smarty->use_include_path && !preg_match('/^([\/\\\\]|[a-zA-Z]:[\/\\\\])/', $_template_dir)) {
                     // try PHP include_path
@@ -429,7 +429,7 @@ class Smarty_Internal_Utility {
         foreach($smarty->getPluginsDir() as $plugin_dir) {
             $_plugin_dir = $plugin_dir;
             $plugin_dir = realpath($plugin_dir);
-            // resolve include_path or fail existance
+            // resolve include_path or fail existence
             if (!$plugin_dir) {
                 if ($smarty->use_include_path && !preg_match('/^([\/\\\\]|[a-zA-Z]:[\/\\\\])/', $_plugin_dir)) {
                     // try PHP include_path
@@ -561,7 +561,7 @@ class Smarty_Internal_Utility {
         foreach($smarty->getConfigDir() as $config_dir) {
             $_config_dir = $config_dir;
             $config_dir = realpath($config_dir);
-            // resolve include_path or fail existance
+            // resolve include_path or fail existence
             if (!$config_dir) {
                 if ($smarty->use_include_path && !preg_match('/^([\/\\\\]|[a-zA-Z]:[\/\\\\])/', $_config_dir)) {
                     // try PHP include_path
diff --git a/gestion/lib/smarty/sysplugins/smarty_resource.php b/gestion/lib/smarty/sysplugins/smarty_resource.php
index 55f1497..0a563df 100644
--- a/gestion/lib/smarty/sysplugins/smarty_resource.php
+++ b/gestion/lib/smarty/sysplugins/smarty_resource.php
@@ -584,7 +584,7 @@ abstract class Smarty_Resource {
  * @author Rodney Rehm
  *
  * @property integer $timestamp Source Timestamp
- * @property boolean $exists    Source Existance
+ * @property boolean $exists    Source Existence
  * @property boolean $template  Extended Template reference
  * @property string  $content   Source Content
  */
@@ -811,7 +811,7 @@ class Smarty_Template_Compiled {
     public $timestamp = null;
 
     /**
-     * Compiled Existance
+     * Compiled Existence
      * @var boolean
      */
     public $exists = false;
diff --git a/gestion/lib/smarty/sysplugins/smarty_security.php b/gestion/lib/smarty/sysplugins/smarty_security.php
index f8b31d9..58f3378 100644
--- a/gestion/lib/smarty/sysplugins/smarty_security.php
+++ b/gestion/lib/smarty/sysplugins/smarty_security.php
@@ -77,7 +77,7 @@ class Smarty_Security {
         'nl2br',
     );
     /**
-     * This is an array of trusted PHP modifers.
+     * This is an array of trusted PHP modifiers.
      *
      * If empty all modifiers are allowed.
      * To disable all modifier set $modifiers = null.
diff --git a/gestion/lib/smarty/version.txt b/gestion/lib/smarty/version.txt
index 9736f8a..f884424 100644
--- a/gestion/lib/smarty/version.txt
+++ b/gestion/lib/smarty/version.txt
@@ -1,5 +1,6 @@
 
 
+28/06/2013 Smarty 3.1.14 Jun 28, 2013
 16/01/2013 Smarty 3.1.13 Jan 15th, 2013
 25/09/2012 Smarty 3.1.12 Sep 25th, 2012
 01/07/2012 Smarty 3.1.11 Jun 30th, 2012
diff --git a/gestion/lib/version_lib_smarty.txt b/gestion/lib/version_lib_smarty.txt
index b3586a4..1d640fe 100644
--- a/gestion/lib/version_lib_smarty.txt
+++ b/gestion/lib/version_lib_smarty.txt
@@ -1,8 +1,9 @@
-http://www.smarty.net/  - http://www.smarty.net/manual/fr/
+﻿http://www.smarty.net/  - http://www.smarty.net/manual/fr/
 http://smarty-php.googlecode.com/svn/trunk/libs/
 
 
 
+28/06/2013 Smarty 3.1.14 Jun 28, 2013  (diff SVN 4694-4747)
 16/01/2013 Smarty 3.1.13 Jan 15th, 2013 (diff SVN 4659-4694)
 25/09/2012 Smarty 3.1.12 Sep 25th, 2012 (diff SVN 4633-4659)
 01/07/2012 Smarty 3.1.11 Jun 30th, 2012  (svn du 01/07 4633)



20/05/2013
----------
c659905b823cd47d02bd4672da2f92c33209bfcb
 gestion/gestassophp_sa/doc/lisez_moi.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gestion/gestassophp_sa/doc/lisez_moi.txt b/gestion/gestassophp_sa/doc/lisez_moi.txt
index 587892b..5b96bc6 100644
--- a/gestion/gestassophp_sa/doc/lisez_moi.txt
+++ b/gestion/gestassophp_sa/doc/lisez_moi.txt
@@ -36,7 +36,7 @@ Configuration nécessaire
 ou
  un accès à une base de données PostgreSQL version 7 ou plus
 - Vérifier sur votre serveur que les fonctionnalité 'magic_quotes_...' sont bien à OFF
- Atention 'magic_quotes_gpc' Cette fonctionnalité est devenue OBSOLETE depuis PHP 5.3.0 et a été SUPPRIMEE depuis PHP 5.4.0.
+ Attention 'magic_quotes_gpc' Cette fonctionnalité est devenue OBSOLETE depuis PHP 5.3.0 et a été SUPPRIMEE depuis PHP 5.4.0.
 
 Vous devez connaître les données de votre connexion MySQL ou PostgreSQL : 
 - l'adresse du serveur MySQL ou PostgreSQL ->si vous êtes en local 'localhost'  


20/05/2013
---------
ac190158a48aa161e86fd61783b769e0276b427f
 gestion/gestassophp_sa/doc/Changelog.txt | 4 ++--
 gestion/gestassophp_sa/doc/lisez_moi.txt | 9 +++++----
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/gestion/gestassophp_sa/doc/Changelog.txt b/gestion/gestassophp_sa/doc/Changelog.txt
index cb76dac..605de72 100644
--- a/gestion/gestassophp_sa/doc/Changelog.txt
+++ b/gestion/gestassophp_sa/doc/Changelog.txt
@@ -12,12 +12,12 @@
  * @package   GestAssoPhp+Pg
  */
 
- 6.5.0 02/05/2013 en cours
+ 6.5.0 20/05/2013 en cours
  -----------------------
  - Modification en base de donnée du champ CP int(8) par VARCHAR( 8 ) pour  les Codes Postaux avec 0 comme 1er chiffre
 	Faites pointer votre navigateur sur l'URL d'installation (ex : http://votreserveur/gestion_s/gestassophp_s/admin/upgrade640-650.php) 
 	une page s'affiche ... "MODIFICATION BASE DE DONNEES " , vérifier si tous les messages sont OK, puis "cliquez sur le lien en bas de page
- 
+ - Vérifier sur votre serveur que les fonctionnalité 'magic_quotes_...' sont bien à OFF.
  
 6.4.0 05/04/2013 Finale
 -----------------------
diff --git a/gestion/gestassophp_sa/doc/lisez_moi.txt b/gestion/gestassophp_sa/doc/lisez_moi.txt
index 322a9fa..587892b 100644
--- a/gestion/gestassophp_sa/doc/lisez_moi.txt
+++ b/gestion/gestassophp_sa/doc/lisez_moi.txt
@@ -29,13 +29,14 @@ Connaitre PHP, Smarty et adodb
 
 Configuration nécessaire
 ************************
-Vous devez au minimum disposer d'un espace sur un serveur Web avec : 
+- Vous devez au minimum disposer d'un espace sur un serveur Web avec : 
  un accès FTP pour l'installation des fichiers, 
-le support de PHP 5.1.x (normalement pour Smarty PHP 5.2)  ou supérieur avec le support des sessions,
+- Le support de PHP 5.1.x (normalement pour Smarty PHP 5.2)  ou supérieur avec le support des sessions,
  un accès à une base de données MySQL version 5 minimum.
 ou
  un accès à une base de données PostgreSQL version 7 ou plus
-
+- Vérifier sur votre serveur que les fonctionnalité 'magic_quotes_...' sont bien à OFF
+ Atention 'magic_quotes_gpc' Cette fonctionnalité est devenue OBSOLETE depuis PHP 5.3.0 et a été SUPPRIMEE depuis PHP 5.4.0.
 
 Vous devez connaître les données de votre connexion MySQL ou PostgreSQL : 
 - l'adresse du serveur MySQL ou PostgreSQL ->si vous êtes en local 'localhost'  
@@ -184,5 +185,5 @@ C - MISE A JOUR DEPUIS UNE VERSION 6.4.0 vers 6.5.x
 
 	
 ----------------------
-Mise à jour 02/05/2013
+Mise à jour 20/05/2013
 


02/05/2013
---------
66544831ab861f702d80001d0032f6ee3db75ce2
 gestion/gestassophp_sa/admin/upgrade640-650.php | 102 ++++++++++++++++++++++++
 gestion/gestassophp_sa/config/version.php       |   2 +-
 gestion/gestassophp_sa/doc/Changelog.txt        |   6 ++
 gestion/gestassophp_sa/doc/lisez_moi.txt        |  18 +++--
 gestion/gestassophp_sa/install/schema.php       |   5 +-
 5 files changed, 124 insertions(+), 9 deletions(-)

diff --git a/gestion/gestassophp_sa/admin/upgrade640-650.php b/gestion/gestassophp_sa/admin/upgrade640-650.php
new file mode 100644
index 0000000..d05d744
--- /dev/null
+++ b/gestion/gestassophp_sa/admin/upgrade640-650.php
@@ -0,0 +1,102 @@
+<?php
+/**
+ * Projet : gestassophp_sa [GestAssoPhp+Pg]
+ * ---------------------------
+ * Licence Creative Commons selon les termes du présent contrat appelé Contrat Public Creative Commons 
+ * Auteur original : Jean-Claude Etiemble
+ * @link :  http://creativecommons.org/licenses/by-sa/2.0/fr/  - Paternité - Partage à l'Identique 2.0 France (CC BY-SA 2.0)
+ * ---------------------------
+ *	
+ * @author : JC Etiemble - http://jc.etiemble.free.fr
+ * @version :  2013
+ * @copyright 2007-2013  (c) JC Etiemble
+ * @package   GestAssoPhp+Pg
+ */
+ 
+/**
+ *  Directory :  /ROOT_DIR_GESTASSO/admin/
+ *   Fichier : upgrade5-sa.php
+ *   MISE A JOUR ENTRE V 5.x et GestAssoPhp+Pg (gestassophp_sa)
+*/
+
+error_reporting(0);
+
+$valid_modif='';
+
+include_once '../config/connexion.php';
+	// la date du jour	
+	$date_du_jour=date('d/m/Y');//Pour définir la date
+		
+
+		
+//------------------------- DEBUT UPGRADE + MODIFICATIONS -------------------
+echo '
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
+ "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="fr">
+<head>
+    <title>GestAssoPhp+Pg mise à jour de V 6.4.0 vers 6.5.0(gestassophp_sa)</title>
+    <meta http-equiv="Content-Type" content="text/HTML; charset={language name=charset}" />		
+	<link rel="stylesheet" type="text/css" media="screen"  href="../js/style_screen.css"/>
+</head>
+<body> 
+';
+
+    echo 'GestAssoPhp+Pg mise &agrave; jour de V 6.4.0 GestAssoPhp vers '.$VERSION.' (gestassophp_sa) le : '.$date_du_jour.'  <br /><br />';
+	
+	echo '<big>MODIFICATION BASE DE DONNEES </big><br />';
+	echo '------------------------------------------- <br />';
+	echo '<b>Modifications Champs cp_adht dans table '.TABLE_ADHERENTS.'</b><br />pour afficher le code postal commen&ccedil;ant par 0 <br />';	
+//	ALTER TABLE `gs_adherent` CHANGE `cp_adht` `cp_adht` VARCHAR( 8 ) NOT NULL DEFAULT '0'
+// POUR mySql
+//  ALTER TABLE "gsa_adherent" ALTER COLUMN "cp_adht" TYPE character varying(8)
+// Pour Postrgresql  integer -> character varying(8)
+//	Colonne		Type			
+// 	cp_adht		character varying(8
+
+	//Si Mysql
+	if ( TYPE_BD_AODB == 'mysql' ) {
+		$req_upgrade_adht = "ALTER TABLE ".TABLE_ADHERENTS." CHANGE `cp_adht` `cp_adht` VARCHAR( 8 ) NOT NULL DEFAULT '0' ";
+		echo $req_upgrade_adht .' <br />';	
+		echo ' <br />';	
+		$dbresult = $db->Execute($req_upgrade_adht);	
+			if ($dbresult == true) {
+				echo 'Mysql - Modification r&eacute;ussie : champ "cp_adht" <br />';
+				$valid_modif = 1;
+			} else {
+				echo '<span class="erreur-Jaunerouge">Erreur requ&ecirc;te ou champ "cp_adht" ! </span><br />';
+				$valid_modif = 0;
+			}
+	}
+	
+	//Si postgresql
+	if ( TYPE_BD == 'postgres' ) {	
+		$req_upgrade_adht = "ALTER TABLE ".TABLE_ADHERENTS." ALTER COLUMN cp_adht TYPE character varying(8) ";
+		echo $req_upgrade_adht .' <br />';	
+		echo ' <br />';	
+		$dbresult = $db->Execute($req_upgrade_adht);	
+			if ($dbresult == true) {
+				echo 'PostgreSql - Modification r&eacute;ussie : champ "cp_adht" <br />';
+				$valid_modif = 1;
+			} else {
+				echo '<span class="erreur-Jaunerouge">Erreur requ&ecirc;te ou champ "cp_adht" ! </span><br />';
+				$valid_modif = 0;
+			}	
+	}
+
+		//echo ' <br />'.$valid_modif;			
+		if ($valid_modif == 1) {
+			echo ' Vous pouvez vous connectez &agrave; <a href="../index.php" title="Connexion">GestAssoPhp+Pg</a>, et v&eacute;rifier le fonctionnement dans Administration<br />';
+		} else {
+			echo '<span class="erreur-Jaunerouge">Erreur requ&ecirc;te - Contacter votre administrateur syst&egrave;me! </span><br />';
+		}
+
+
+
+
+echo '  
+  </body>
+</html>
+';
+
+?>
\ No newline at end of file
diff --git a/gestion/gestassophp_sa/config/version.php b/gestion/gestassophp_sa/config/version.php
index 5aaef14..4a60c06 100644
--- a/gestion/gestassophp_sa/config/version.php
+++ b/gestion/gestassophp_sa/config/version.php
@@ -21,7 +21,7 @@
 */
 
 /**** La version de GestAsssoPhp  */
-$VERSION = "GestAssoPhp+Pg V 6.4.0";
+$VERSION = "GestAssoPhp+Pg V 6.5.0";
 define('VERSION', $VERSION);
 
 
diff --git a/gestion/gestassophp_sa/doc/Changelog.txt b/gestion/gestassophp_sa/doc/Changelog.txt
index 00d5a51..cb76dac 100644
--- a/gestion/gestassophp_sa/doc/Changelog.txt
+++ b/gestion/gestassophp_sa/doc/Changelog.txt
@@ -12,6 +12,12 @@
  * @package   GestAssoPhp+Pg
  */
 
+ 6.5.0 02/05/2013 en cours
+ -----------------------
+ - Modification en base de donnée du champ CP int(8) par VARCHAR( 8 ) pour  les Codes Postaux avec 0 comme 1er chiffre
+	Faites pointer votre navigateur sur l'URL d'installation (ex : http://votreserveur/gestion_s/gestassophp_s/admin/upgrade640-650.php) 
+	une page s'affiche ... "MODIFICATION BASE DE DONNEES " , vérifier si tous les messages sont OK, puis "cliquez sur le lien en bas de page
+ 
  
 6.4.0 05/04/2013 Finale
 -----------------------
diff --git a/gestion/gestassophp_sa/doc/lisez_moi.txt b/gestion/gestassophp_sa/doc/lisez_moi.txt
index 1a8f907..322a9fa 100644
--- a/gestion/gestassophp_sa/doc/lisez_moi.txt
+++ b/gestion/gestassophp_sa/doc/lisez_moi.txt
@@ -55,7 +55,7 @@ error_reporting = E_ALL & ~E_DEPRECATED
 date.timezone = Europe/Paris
 
 
-
+-------------------------
 A - INSTALLATION NOUVELLE
 -------------------------
 1. Vous avez récupérer l'archive gestion_saxx.tar.bz2
@@ -153,7 +153,9 @@ Il est conseillé de faire la mise à jour, pour cela utiliser les fichiers de m
 Pour information les modifications entre la version n et la version n+1 sont indiquées dans le fichier Diff_6xxx_6xxx.diff
 
 
-B - MISE A JOUR DEPUIS UNE VERSION 5.x (mini 5.3) 
+-----------------------------------------------------
+B - MISE A JOUR DEPUIS UNE VERSION 5.x (mini 5.3)
+----------------------------------------------------- 
 CONSULTER le fichier maj_lire.txt
 
 
@@ -173,10 +175,14 @@ pour la mise à jour pendre les précautions d'usage
 2- envoyer les nouveaux fichiers de l'archive version 6.x en FTP 
 3- Se loguer en mode administrateur (niveau 9)
 
+-----------------------------------------------------
+C - MISE A JOUR DEPUIS UNE VERSION 6.4.0 vers 6.5.x  
+-----------------------------------------------------
+ - Modification en base de donnée du champ CP int(8) par VARCHAR( 8 ) pour  les Codes Postaux avec 0 comme 1er chiffre
+	Faites pointer votre navigateur sur l'URL d'installation (ex : http://votreserveur/gestion_s/gestassophp_s/admin/upgrade640-650.php) 
+	une page s'affiche ... "MODIFICATION BASE DE DONNEES " , vérifier si tous les messages sont OK, puis "cliquez sur le lien en bas de page
 
-
-
-
+	
 ----------------------
-Mise à jour 06/02/2013
+Mise à jour 02/05/2013
 
diff --git a/gestion/gestassophp_sa/install/schema.php b/gestion/gestassophp_sa/install/schema.php
index d5cd8de..c13be38 100644
--- a/gestion/gestassophp_sa/install/schema.php
+++ b/gestion/gestassophp_sa/install/schema.php
@@ -76,8 +76,9 @@ l'UTF-8, qui permet théoriquement d'encoder toutes les langues, du français au
 		$taboptarray = array();
 
 
+
 //*************************
-#--  1 Structure de la table `_adherent`	
+#--  1 Structure de la table `_adherent`	 // modifié le 02/05/2013 pour CP avec 0 comme 1er chiffre
 	$flds = "
 		id_adht I(4) KEY AUTO, 
 		soc_adht  C(3) NOT NULL default '',
@@ -85,7 +86,7 @@ l'UTF-8, qui permet théoriquement d'encoder toutes les langues, du français au
 		prenom_adht  C(50) NOT NULL default '',
 		nom_adht  C(50) NOT NULL default '',
 		adresse_adht  C(100) default NULL,
-		cp_adht  I(8) NOT NULL default '0',
+		cp_adht  C(8) NOT NULL default '',
 		ville_adht  C(50) NOT NULL default '',
 		telephonef_adht  C(16) default NULL,
 		telephonep_adht C(16) default NULL,
